gammapy/visualization/utils.py

```diff
@@ -3,32 +3,67 @@
 __all__ = ["plot_spectrum_datasets_off_regions", "plot_contour_line"]
 
 
-def plot_spectrum_datasets_off_regions(datasets, ax=None):
-    """Plot spectrum datasets of regions.
+def plot_spectrum_datasets_off_regions(datasets, ax=None, legend=None, **kwargs):
+    """Plot spectrum datasets' off regions.
 
     Parameters
     ----------
-    datasets : list of `SpectrumDatasetOnOff`
-        List of spectrum on-off datasets
+    datasets : `Datasets` of or list of `SpectrumDatasetOnOff`
+        List of spectrum on-off datasets.
+    ax : `~`
+        .
+    legend : bool
+        Whether to display the legend. By default True if ``len(datasets) <= 10``.
+    **kwargs : dict
+        Keyword arguments used in `gammapy.maps.RegionNDMap.plot_region`.
+        Can contain a `~cycler.Cycler` in a ``prop_cycle`` item.
+
+    Notes
+    -----
+    Properties from the ``prop_cycle`` have maximum priority, except ``color``.
+    ``edgecolor``/``color`` is selected from the sources below in this order:
+        ``kwargs["edgecolor"]``
+        ``kwargs["prop_cycle"]``
+        ``matplotlib.rcParams["axes.prop_cycle"]``
+        ``matplotlib.rcParams["patch.edgecolor"]``
+    ``matplotlib.rcParams["patch.facecolor"]`` is never used.
+
+    Examples
+    --------
+    >>> plot_spectrum_datasets_off_regions(datasets, ax, legend=False, lw=2.5)
+    >>> plot_spectrum_datasets_off_regions(datasets, ax, alpha=0.3, facecolor='k')
+    >>> plot_spectrum_datasets_off_regions(
+            datasets, ax, ls='--', prop_cycle=plt.cycler('color', list('rgb'))
+        )
+    >>> plt.rc('legend', fontsize=9)
+    >>> plt.rc('patch', edgecolor='blue')
+    >>> plot_spectrum_datasets_off_regions(
+            datasets, ax, legend=True, prop_cycle=plt.cycler('ls', ['-', '-.'])
```

Comment:
Maybe modify the example here to cycle both through the color and lines style? This might be even more helpful...

Reply:
In this last example I wanted to show that it is possible to not cycle edgecolors (if the user wants to have a specific color for different `Datasets` in the same `ax`, for example). However, it would be inappropriate to set each static `edgecolor` through rcParams. Therefore, I will improve it and make my intentions clearer by grouping this example with a second function call. I'll also boost the previous example with a `color` + `ls` cycle.